addon/components/ember-collection.js

```diff
@@ -101,20 +100,52 @@ export default Ember.Component.extend({
 
   updateScrollPosition(){
     if (!this._scrollChange) { return; } // don't process bound scroll coords unless our action is being handled
-    let scrollLeftAttr = this.getAttr('scroll-left');
-    if (scrollLeftAttr !== undefined) {
-      scrollLeftAttr = parseInt(scrollLeftAttr, 10);
-      if (this._scrollLeft !== scrollLeftAttr) {
-        this.set('_scrollLeft', scrollLeftAttr);
-      }
+    let scrollIndexAttr = this.getAttr('scroll-index');
+    if (scrollIndexAttr !== undefined) {
+      scrollIndexAttr = parseInt(scrollIndexAttr, 10);
     }
+    if (this._scrollIndex !== scrollIndexAttr && (this._clientWidth || this._clientHeight)) {
```

Comment:
this area is quite complicated, it may be worth extracting to a function so it is easier to unit test.